Lib/test/test_contextlib.py

```diff
@@ -211,6 +212,21 @@ def woohoo(self, func, args, kwds):
         with woohoo(self=11, func=22, args=33, kwds=44) as target:
             self.assertEqual(target, (11, 22, 33, 44))
 
+    def test_nokeepref(self):
+        class A:
+            pass
+
+        @contextmanager
+        def whohoo(a, b):
```

Comment:
s/whohoo/woohoo/ (Yes, it's just a silly name for a throwaway function, but we may as well spell it correctly and consistently)